pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -666,26 +666,40 @@ public SuccessResponse deleteSegment(
   @Path("/segments/{tableName}")
   @Authorize(targetType = TargetType.TABLE, paramName = "tableName", action = Actions.Table.DELETE_SEGMENT)
   @Authenticate(AccessType.DELETE)
-  @ApiOperation(value = "Delete all segments", notes = "Delete all segments")
-  public SuccessResponse deleteAllSegments(
+  @ApiOperation(value = "Delete the list of segments provided in the payload else all segments",
+      notes = "Delete the list of segments provided in the payload else all segments")
+  public SuccessResponse deleteMultipleSegments(
       @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
       @ApiParam(value = "OFFLINE|REALTIME", required = true) @QueryParam("type") String tableTypeStr,
       @ApiParam(value = "Retention period for the table segments (e.g. 12h, 3d); If not set, the retention period "
           + "will default to the first config that's not null: the table config, then to cluster setting, then '7d'. "
           + "Using 0d or -1d will instantly delete segments without retention")
-      @QueryParam("retention") String retentionPeriod, @Context HttpHeaders headers) {
+      @QueryParam("retention") String retentionPeriod,
+      @ApiParam(value = "Segment names to be deleted if not provided deletes all segments by default",
+          allowMultiple = true) @QueryParam("segmentNames") List<String> segments, @Context HttpHeaders headers) {
```

Comment:
Suggest renaming it to be aligned with `columns` in other APIs
```suggestion
          allowMultiple = true) @QueryParam("segments") List<String> segments, @Context HttpHeaders headers) {
```

Reply:
^^ (Maybe you haven't pushed the change yet)

Reply:
committed this